x-pack/platform/plugins/shared/streams_app/public/hooks/use_timefilter.tsx

```diff
@@ -5,64 +5,22 @@
  * 2.0.
  */
 
-import React, { useContext, useState, useMemo, useCallback } from 'react';
-import { TimeRange } from '@kbn/es-query';
-import { TimefilterContract } from '@kbn/data-plugin/public';
-import useObservable from 'react-use/lib/useObservable';
-import { map, tap } from 'rxjs';
-
-const TimeFilterContext = React.createContext<TimeFilterState | undefined>(undefined);
-
-export interface TimeFilterProviderProps {
-  timefilter: TimefilterContract;
-  children?: React.ReactNode;
-}
-
-export interface AbsoluteTimeRange {
-  start: number;
-  end: number;
-}
-
-export interface TimeFilterState {
-  timeRange: TimeRange;
-  absoluteTimeRange: AbsoluteTimeRange;
-  setTimeRange(value: TimeRange): void;
-  refreshAbsoluteTimeRange(): void;
-}
-
-export const TimeFilterProvider: React.FC<TimeFilterProviderProps> = ({ timefilter, children }) => {
-  const [absoluteTimeRange, setAbsoluteTimeRange] = useState(timefilter.getAbsoluteTime());
-
-  const refreshAbsoluteTimeRange = useCallback(
-    () => setAbsoluteTimeRange(timefilter.getAbsoluteTime()),
-    [timefilter]
-  );
-
-  const timeRange$ = useMemo(() => {
-    return timefilter.getTimeUpdate$().pipe(map(timefilter.getTime), tap(refreshAbsoluteTimeRange));
-  }, [timefilter, refreshAbsoluteTimeRange]);
-
-  const timeRange = useObservable(timeRange$, timefilter.getTime());
-
-  const state: TimeFilterState = useMemo(() => {
-    return {
-      timeRange,
-      absoluteTimeRange: {
-        start: new Date(absoluteTimeRange.from).getTime(),
-        end: new Date(absoluteTimeRange.to).getTime(),
+import type { TimefilterContract, TimefilterHook } from '@kbn/data-plugin/public';
+import { useKibana } from './use_kibana';
+
+export function useTimefilter(): TimefilterHook & { setTime: TimefilterContract['setTime'] } {
```

Comment:
Why do we still need our own Streams specific hook for this if the shared one has been fixed to handle absolute time correctly?

Reply:
the shared hook doesn't have a setTime() function. I've added it here for convenience. We could also use the useTimefilter hook directly as setTime() is colocated with the hook, but it's a lot of destructuring for a popular use case.

Reply:
Would it be possible to add the setTime function to the shared hook instead of creating ap project specific one? 

Would be good to have it consistent I think.

Reply:
Hmm at that point I would rather not have it - because there's also other state-changing methods on timefilter that can be useful. I guess I'm optimizing for the most common use cases, but I'm also happy to remove entirely if you feel strongly about it.

Reply:
I dunno, for me get and set is kind of the most basic/common functionality so I wouldn't see that as over polluting the API surface of this hook. (like `const [value, setValue] = useState()`, it doesn't get more simple than that I think)

Curious what other people think.